### PR TITLE
Syndicate bomb delay wire changed to one time use

### LINKED ISSUE #16774: Pulsing syndicate bomb wires has no cooldown.
I think we can all extrapolate why this is a problem

### PR DESCRIPTION
**Pull request by LatD**
Pulsing is only possible once now and the time gained from it is changed from 10 seconds to 30 seconds.
 ~~Fixes  #16774~~ nevermind, it's apparently not related directly to this

~~I hope that variable thing works correctly. I'm not sure if it can be added to this wire file like that or to the bomb file as a bomb variable.~~ Fixed

:cl: Lati
tweak: Syndicate bomb can be only delayed once by pulsing the delay wire. Time gained from this is changed from 10 seconds to 30 seconds.
/:cl:

### CODE REVIEW

**Comment by Core0verload** [comment]
Don't like it. Just add a cap that will prevent you from delaying explosion into 99999999999 seconds. If the delay timer is capped at 60, it would take a lot of constant pulsing to make it not explode.

**Comment by Supermichael777** [comment]
"Did you defuse the bomb?"
"Nah i just set the timer for the heat death of the universe. Should be fine."

**Comment by LatD** [comment]
Capping the timer to 60 seconds would be another easy way of doing this. That would allow it to be endlessly delayed as long as someone stays there and keeps pulsing.

**Comment by Supermichael777** [comment]
the point is the bomb must eventualy go off and a 60 second timer cap means that it never will and you can get the bomb squad "eventualy"

**Comment by bgobandit** [comment]
:-1:

the point of that issue was people can exploit the wire that ADVANCES the bomb timer

**Comment by coiax** [comment]
bgo, if they wanted to advance the bomb timer, then wouldn't they just, you know, pulse the wire that makes it explode?

**Comment by bgobandit** [comment]
not if they want to live

**Comment by coiax** [comment]
Oh, I see. I was thinking from the perspective of pulsing the wires remotely. But your concern is that it's too easy to slap down a bomb, open it, determine which wire is the advance clock wire, arm the bomb, then pulse it until the countdown is something short, and then run.

**Comment by LatD** [comment]
How is that an exploit if the only thing it does is kill the user? The bomb will set the timer to 10 seconds if that wire is pulsed with less than 11 seconds anyway.

**Comment by Lumipharon** [comment]
I actually don't understand the issue here, other than having no maximum limit on the timer?

If you're trying to make it explode early, then you can already find the boom wire with 100% safetly, to remotely detonate.

If you're trying to defuse, as long as a max timer gets added, you still have to find the disarm wire eventually.

**Comment by KazeEspada** [comment]
Or space it.

**Comment by Cheridan** [comment]
>I hope that variable thing works correctly. I'm not sure if it can be added to this wire file like that or to the bomb file as a bomb variable.

Did you test it

**Comment by LatD** [comment]
Coders testing their code, honk! This didn't work with the wire variable. I changed it to bomb variable, now it's tested to work. 
I also noticed that the deactivation wire does the same thing so I did the same thing to it too with a higher timer. That shouldn't matter much since if they can find that wire they can deactivate the bomb properly anyway.

File: code/datums/wires/syndicatebomb.dm [hunk 1]
```diff
@@ -42,9 +46,12 @@
 				playsound(B, 'sound/machines/click.ogg', 30, 1)
 				B.active = TRUE
 				B.update_icon()
+			if(B.delayedlittle)
+				holder.visible_message("<span class='notice'>\icon[B] Nothing happens.</span>")
 			else
 				holder.visible_message("<span class='notice'>\icon[B] The bomb seems to hesitate for a moment.</span>")
-				B.timer += 5
+				B.timer += 10
+				B.delayedlittle = TRUE
```
Comment (phil235) [inline code/datums/wires/syndicatebomb.dm]: With the current code, if I pulse this wire on a bomb that is neither active or defused, I will get both "the bomb start ticking!" and " the bomb seems to hesitate".

the logic should be "if(!B.active && !B.defused) XXXX else if(B.delaylittle) XXXX else XXXXX

Comment (LatD) [inline code/datums/wires/syndicatebomb.dm]: Thanks, fixed now. It's pretty rare situation so I hadn't even thought about it.
Should I add a similar check to the delay wire so bigdelay cannot be activated before even arming the bomb or shall we use it like this?

Comment (phil235) [inline code/datums/wires/syndicatebomb.dm]: It's alright like that. You can add an if(active) for bigdelay if you want but I don't think anyone will mind if it's not there.

**Pull request merged**